pyam/core.py

```diff
@@ -779,7 +779,11 @@ def timeseries(self, iamc_index=False):
                 )
             s = s.droplevel(self.extra_cols)
 
-        return s.unstack(level=self.time_col).rename_axis(None, axis=1)
+        # calling `unstack` on the `s` directly runs into MemoryError for large datasets
+        cols = [i for i in s.index.names if i != self.time_col]
+        return pd.concat(
+            [_data.unstack(level=self.time_col) for i, _data in s.groupby(level=cols)]
```

Comment:
One small improvement could be to use `_` instead of `i` in the loop to follow the convention for unused iteration variables. 
```suggestion
            [_data.unstack(level=self.time_col) for _, _data in s.groupby(level=cols)]
```